sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/event_processor.py

```diff
@@ -235,6 +246,7 @@ def _create_tasks_for_claimed_ownership(
             ) = self.get_init_event_position(partition_id, checkpoint)
             if partition_id in self._partition_contexts:
                 partition_context = self._partition_contexts[partition_id]
+                partition_context._last_received_event = None  # pylint:disable=protected-access
```

Comment:
Isn't this already set to None in the partition context initialized?
Under what circumstances would this not be set?

Reply:
This is for re-balancing. When a client loses a partition and get back the partition some time later because of partition re-balancing, the partition_context isn't recreated. So reset _last_received_event=None here.